### PR TITLE
AAE-25816 Allow to set a custom storage prefix for Authentication

### PR DESCRIPTION
**Pull request by alep85**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/auth/services/custom-auth-storage.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Inject, Injectable, InjectionToken } from '@angular/core';
+import { StorageService } from '../../common/services/storage.service';
+import { Observable } from 'rxjs';
+
+export const CUSTOM_AUTH_STORAGE_PREFIX = new InjectionToken<any>('CUSTOM_AUTH_STORAGE_PREFIX');
```
Comment (DenysVuika) [inline lib/core/src/lib/auth/services/custom-auth-storage.service.ts:22]: there's already another hack for that introduced by HxP earlier

```
export const STORAGE_PREFIX_FACTORY_SERVICE = new InjectionToken<StoragePrefixFactoryService>('STORAGE_PREFIX_FACTORY_SERVICE');

@Injectable({
    providedIn: 'root'
})
export class StoragePrefixFactory {
    constructor(
        private appConfigService: AppConfigService,
        @Optional()
        @Inject(STORAGE_PREFIX_FACTORY_SERVICE) private storagePrefixFactory?: StoragePrefixFactoryService
    ) {}

```

**Comment by DenysVuika** [comment]
@alep85 please see the `StoragePrefixFactory` introduced by @BSekula

File: lib/core/src/lib/auth/services/custom-auth-storage.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            providers: [
+                CustomAuthStorageService,
+                { provide: CUSTOM_AUTH_STORAGE_PREFIX, useValue: of('custom-prefix') }
+            ]
+        });
+        service = TestBed.inject(CustomAuthStorageService);
+    });
+
+    it('should be created', () => {
+        expect(service).toBeTruthy();
```
Comment (DenysVuika) [inline lib/core/src/lib/auth/services/custom-auth-storage.service.spec.ts:37]: this test can be removed

Comment (alep85) [reply lib/core/src/lib/auth/services/custom-auth-storage.service.spec.ts:37]: will do

**Comment by alep85** [comment]
Closed since it has been handled by another [PR](https://github.com/Alfresco/hxp-frontend-apps/pull/9271)

**Pull request closed without merging**